sdk/src/androidTest/java/com/mapbox/maps/StyleLoadTest.kt

```diff
@@ -169,6 +172,53 @@ class StyleLoadTest {
     assertEquals(loadedStyles, 1)
   }
 
+  @Test
+  fun testTerrainDoesNotCrashOnMultipleStyleLoad() {
+    fun loadStyleWithTerrain(
+      uri: String = Style.DARK,
+      terrainSource: String = "TERRAIN_SOURCE",
+      demSourceUri: String = "mapbox://mapbox.mapbox-terrain-dem-v1"
+    ) {
+      mapboxMap.loadStyle(
```

Comment:
nit:
maybe add extra delay? potentially we can have super fast internet and super fast device.

Reply:
Well worst case if it happens without any delay - then the crashes actually had happenned. Could you elaborate more why do you think delay is needed?

Reply:
>Could you elaborate more why do you think delay is needed?

to guarantee a delay between different styles load

Reply:
But we don't need a delay, that's the thing, for this specific test - it happens when there is no delay so that style load events could collide with higher probability

Reply:
I mean some delay in style loaded callbacks. To make listeners crossover effect more noticeable.

Reply:
But I can add bunch of randomly delayed calls, all right, no problem

Reply:
Added 100 launches with random delay of 0-5 millis